### PR TITLE
Merge release-v4.9 branch

### PR DESCRIPTION
**Pull request by ernestognw**
Merging changes from release-v4.9

### CODE REVIEW

**Comment by Amxx** [comment]
Most of the changeset removal are part of 5.0 and should be kept

File: test/governance/Governor.test.js [hunk 1]
```diff
@@ -701,6 +701,89 @@ contract('Governor', function (accounts) {
         });
       });
 
+      describe('frontrun protection using description suffix', function () {
```
Comment (frangio) [inline test/governance/Governor.test.js:704]: I think these tests are now twice in the JS file.

**Comment by ernestognw** [comment]
The merge resolution was kind of difficult. My conclusion is that we can't correctly automate the merge back of this because there are multiple cases in which we need manual resolution and I actually think it's dangerous to even keep this last release cycle automation.

We can discuss it later. Currently is ready for review.

File: CHANGELOG.md [hunk 2]
```diff
@@ -55,10 +55,14 @@ Users that were registering EIP-165 interfaces with `_registerInterface` from `E
 
 ```solidity
 function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
-    return interfaceId == type(MyInterface).interfaceId || super.supportsInterface(interfaceId);
+  return interfaceId == type(MyInterface).interfaceId || super.supportsInterface(interfaceId);
```
Comment (frangio) [inline CHANGELOG.md:58]: Not sure where this change came from it seems unrelated to the merge.

```suggestion
    return interfaceId == type(MyInterface).interfaceId || super.supportsInterface(interfaceId);
```

Comment (ernestognw) [reply CHANGELOG.md:58]: It went from formatting the `CHANGELOG.md` after conflict resolution

**Review by frangio — APPROVED** [review]
LGTM
